hpedockerplugin/hpe/hpe_3par_mediator.py

```diff
@@ -0,0 +1,1032 @@
+    Version history:
+        1.0.0 - Begin Liberty development (post-Kilo)
+        1.0.1 - Report thin/dedup/hp_flash_cache capabilities
+        1.0.2 - Add share server/share network support
+        1.0.3 - Use hp3par prefix for share types and capabilities
+        2.0.0 - Rebranded HP to HPE
+        2.0.1 - Add access_level (e.g. read-only support)
+        2.0.2 - Add extend/shrink
+        2.0.3 - Fix SMB read-only access (added in 2.0.1)
+        2.0.4 - Remove file tree on delete when using nested shares #1538800
+        2.0.5 - Reduce the fsquota by share size
+                when a share is deleted #1582931
+        2.0.6 - Read-write share from snapshot (using driver mount and copy)
+        2.0.7 - Add update_access support
+        2.0.8 - Multi pools support per backend
+        2.0.9 - Fix get_vfs() to correctly validate conf IP addresses at
+                boot up #1621016
+    """
+
+    VERSION = "2.0.9"
+
+    def __init__(self, host_config, config):
+        self._host_config = host_config
+        self._config = config
+        self._client = None
+        self.client_version = None
+
+    @staticmethod
+    def no_client():
+        return hpe3parclient is None
+
+    def do_setup(self, timeout=30):
+
+        if self.no_client():
+            msg = _('You must install hpe3parclient before using the 3PAR '
+                    'driver. Run "pip install --upgrade python-3parclient" '
+                    'to upgrade the hpe3parclient.')
+            LOG.error(msg)
+            raise exception.HPE3ParInvalidClient(message=msg)
+
+        self.client_version = hpe3parclient.version_tuple
+        if self.client_version < MIN_CLIENT_VERSION:
+            msg = (_('Invalid hpe3parclient version found (%(found)s). '
+                     'Version %(minimum)s or greater required. Run "pip'
+                     ' install --upgrade python-3parclient" to upgrade'
+                     ' the hpe3parclient.') %
+                   {'found': '.'.join(map(six.text_type, self.client_version)),
+                    'minimum': '.'.join(map(six.text_type,
+                                            MIN_CLIENT_VERSION))})
+            LOG.error(msg)
+            raise exception.HPE3ParInvalidClient(message=msg)
+
+        try:
+            self._client = file_client.HPE3ParFilePersonaClient(
+                self._config.hpe3par_api_url)
+        except Exception as e:
+            msg = (_('Failed to connect to HPE 3PAR File Persona Client: %s') %
+                   six.text_type(e))
+            LOG.exception(msg)
+            raise exception.ShareBackendException(message=msg)
+
+        try:
+            ssh_kwargs = {}
+            if self._config.hpe3par_san_ssh_port:
+                ssh_kwargs['port'] = self._config.hpe3par_san_ssh_port
+            if self._config.ssh_conn_timeout:
+                ssh_kwargs['conn_timeout'] = self._config.ssh_conn_timeout
+            if self._config.hpe3par_san_private_key:
```

Comment:
@imran-ansari please check this parameter, should it be san_private_key ?

Reply:
This is coming from the implementation where we have two configuration files. Now that we have single configuration file, we will stick to the name "san_private_key" that has been used in volume implementation.